contracts/FriendGroupMarketFactory.sol

```diff
@@ -0,0 +1,698 @@
+ * @title FriendGroupMarketFactory
+ * @notice Factory for creating small-scale prediction markets between friends
+ * @dev Supports P2P betting with tiered membership and member limits
+ * 
+ * KEY FEATURES:
+ * - Tiered membership system (gas-only markets for members)
+ * - Member limit enforcement to prevent bypassing public markets
+ * - Support for 1v1 bets, group prop bets, and event tracking scenarios
+ * - Optional third-party arbitration
+ * - Integration with RagequitModule for fair exits
+ * 
+ * USE CASES:
+ * 1. Competitive event tracking (poker, board games, etc.)
+ * 2. 1v1 prop bets between friends
+ * 3. Small group predictions with arbitrator
+ * 4. Friend group contests and competitions
+ */
+contract FriendGroupMarketFactory is Ownable, ReentrancyGuard {
+    
+    // Market type to distinguish friend markets from public markets
+    enum MarketType {
+        OneVsOne,           // 1v1 direct bet between two parties
+        SmallGroup,         // 3-10 participants
+        EventTracking,      // Tracking for competitive events/games
+        PropBet             // General proposition bet
+    }
+    
+    struct FriendMarket {
+        uint256 marketId;              // ID in ConditionalMarketFactory
+        MarketType marketType;
+        address creator;
+        address[] members;             // Limited participant list
+        address arbitrator;            // Optional third-party for resolution
+        uint256 memberLimit;           // Max concurrent members
+        uint256 creationFee;           // Reduced fee for friend markets
+        uint256 createdAt;
+        bool active;
+        string description;
+        uint256 peggedPublicMarketId;  // Public market ID to peg resolution to (0 = none)
+        bool autoPegged;               // Whether resolution is pegged to public market
+    }
+    
+    // Friend market ID => FriendMarket
+    mapping(uint256 => FriendMarket) public friendMarkets;
+    
+    // User => array of friend market IDs they're in
+    mapping(address => uint256[]) public userMarkets;
+    
+    // Member count tracking per market
+    mapping(uint256 => uint256) public memberCount;
+    
+    // Track which public markets have pegged friend markets
+    mapping(uint256 => uint256[]) public publicMarketToPeggedFriendMarkets;
+    
+    uint256 public friendMarketCount;
+    
+    // Reference to main market factory
+    ConditionalMarketFactory public marketFactory;
+    
+    // Reference to ragequit module
+    RagequitModule public ragequitModule;
+    
+    // Reference to tiered role manager for membership checks
+    TieredRoleManager public tieredRoleManager;
+    
+    // Pricing tiers (for non-members, legacy support)
+    uint256 public constant PUBLIC_MARKET_FEE = 1 ether;      // Standard market fee
+    uint256 public constant FRIEND_MARKET_FEE = 0.1 ether;    // Reduced fee for friend markets
+    uint256 public constant ONE_V_ONE_FEE = 0.05 ether;       // Even lower for 1v1
```

Comment:
do not hard code values for these contracts, thes should be update able by the fairwins managers. fees shoild also support erc 20 tokens